File: RecipeGuide_CosmosDBNoSQLOpenAICognitiveSearch/Model/Recipe.cs
using Azure.Search.Documents.Indexes.Models;
using Azure.Search.Documents.Indexes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CosmosRecipeGuide
{

    public class Recipe
    {

        public string id { get; set; }
        public string name { get; set; }
        public string description { get; set; }
        public List<float> embedding { get; set; }
        public string cuisine { get; set; }
        public string difficulty { get; set; }
        public string prepTime { get; set; }
        public string cookTime { get; set; }
        public string totalTime { get; set; }
        public int servings { get; set; }
        public List<string> ingredients { get; set; }
        public List<string> instructions { get; set; }
    }     


}

File: Java/CosmosDB-NoSQL-VectorSearch/Code/src/main/java/com/azure/recipe/model/Recipe.java
package com.azure.recipe.model;

import lombok.Data;

import java.util.List;

@Data
public class Recipe {
    public String id;
    public String name;
    public String description;
    public List<Double> embedding;
    public String cuisine;
    public String difficulty;
    public String prepTime;
    public String cookTime;
    public String totalTime;
    public int servings;
    public List<String> ingredients;
    public List<String> instructions;

}
